### DISCUSSION TITLE
Configure different standards per folder

### DISCUSSION THREAD

**Discussion by mafeka** [squizlabs/PHP_CodeSniffer#3863] [Q&A] [answered]
Hey,
is it possible to apply different standards for subfolders, but configure everything in a single config file?

**Example:**
```
/ 
    [Standard: Custom in house standard]
    ./subfolder-1
        [Standard: PSR12]
    ./subfolder-2
        [Standard: Custom in house standard (inherited)]
    ./subfolder-3
         [Standard: PSR12]
```

**Background:**
- Monolithic project, main folder uses outdated code standard, some subfolders use modern PSR12
- Team uses PHPStorm (which only loads a single phpcs.xml file)

For the live of me I can neither get nested standards per multiple phpcs.xml files (one per subfolder) to work, nor have I found a way to configure something like this in the main phpcs.xml

Any guidance would be much appreciated.

**Comment by jrfnl** [comment, marked as answer]
I'm thinking you may be able to achieve this using well-configured `<include-pattern>`s, but I haven't tested it.

Something along the lines of:
```xml
<file>.</file>

<rule ref="Custom">
    <include-pattern>./[^/]+\.php</include-pattern><!-- Files in the project root. -->
    <include-pattern>./subfolder-2/</include-pattern>
</rule>

<rule ref="PSR12">
    <include-pattern>./subfolder-[13]{1}/</include-pattern>
</rule>
```

Note: you won't be able to set different property settings for the same sniff for different folders, so your mileage may vary. (tracked in #2258)


Hope this helps.

**Reply by mafeka** [reply]
Works like a charme and covers our usecase, thanks!
